lib/useSmartLink.ts

```diff
@@ -1,17 +1,23 @@
 import KontentSmartLink from "@kontent-ai/smart-link"
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 
-export const useSmartLink = () => useEffect(() => {
-  const envId = process.env.NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID;
-  if (!envId) {
-    throw new Error("Cannot initialize smart-link sdk without 'NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID' environment variable.");
-  }
-  const sdk = KontentSmartLink.initialize({
-    defaultDataAttributes: {
-      projectId: envId,
-      languageCodename: "default",
+export const useSmartLink = () => {
+  const [sdk, setSdk] = useState<KontentSmartLink | null>(null);
+
+  useEffect(() => {
+    const envId = process.env.NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID;
+    if (!envId) {
+      throw new Error("Cannot initialize smart-link sdk without 'NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID' environment variable.");
     }
-  });
+    setSdk(KontentSmartLink.initialize({
+      defaultDataAttributes: {
+        projectId: envId,
+        languageCodename: "default",
+      }
+    }));
+
+    return () => sdk?.destroy()
+  }, [setSdk, sdk]);
```

Comment:
You don't need to put the `setSdk` into the dependency array. It never changed, even the `eslint` rule doesn't enforce putting it there.